src/js/grapl-cdk/lib/fargate_service.ts

```diff
@@ -0,0 +1,192 @@
+import * as cdk from '@aws-cdk/core';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import * as subscriptions from '@aws-cdk/aws-sns-subscriptions';
+
+import * as ec2 from "@aws-cdk/aws-ec2";
+import * as ecs from "@aws-cdk/aws-ecs";
+import * as sqs from "@aws-cdk/aws-sqs";
+import * as ecs_patterns from "@aws-cdk/aws-ecs-patterns";
+import {ContainerImage} from "@aws-cdk/aws-ecs";
+import {GraplServiceProps} from "./grapl-cdk-stack";
+import {Watchful} from "cdk-watchful";
+
+export class Queues {
+    readonly queue: sqs.Queue;
+    readonly retry_queue: sqs.Queue;
+    readonly deadLetterQueue: sqs.Queue;
+
+    constructor(scope: cdk.Construct, queue_name: string) {
+        this.deadLetterQueue = new sqs.Queue(scope, 'DeadLetterQueue', {
+            queueName: queue_name + '-dead-letter-queue',
+        });
+
+        this.retry_queue = new sqs.Queue(scope, 'RetryQueue', {
+            queueName: queue_name + '-retry-queue',
+            deadLetterQueue: { queue: this.deadLetterQueue, maxReceiveCount: 3 },
+            visibilityTimeout: cdk.Duration.seconds(360),
+        });
+
+        this.queue = new sqs.Queue(scope, 'Queue', {
+            queueName: queue_name + '-queue',
+            deadLetterQueue: { queue: this.retry_queue, maxReceiveCount: 3 },
+            visibilityTimeout: cdk.Duration.seconds(180),
+        });
+    }
+}
+
+
+export interface FargateServiceProps {
+    prefix: string;
+    version: string;
+    readsFrom: s3.IBucket;
+    subscribesTo: sns.ITopic;
```

Comment:
These two are sort of 'tied together' like the EventEmitter, right? if so, I'd prefer to see them grouped in 1 object, or at least have similar names (readsFromTopic?)